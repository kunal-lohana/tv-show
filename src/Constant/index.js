export const Constants = {
  NAV_Gener: "Genre",
  CAST: "Cast",
  EPISODES: "Episodes",
  SHOW_DETAILS_BTN: "Show Details",
  OFFICIAL_SITE: "Offical Site",
  DESCRIPTION: "Description",
  SCHEDULE_AVAILABILITY: "Schedule",
  PAGE_NOT_FOUND: "PAGE NOT FOUND",
  ERROR_CODE: "404",
  HOME_PAGE: "Home"
};
