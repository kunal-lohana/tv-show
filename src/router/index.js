import Vue from "vue";
import VueRouter from "vue-router";
import TvShowList from "@/components/TvShowList/TvShowList.vue";
import ShowDetail from "@/components/ShowDetail/ShowDetail.vue";
import PageNotFound from "@/components/PageNotFound";

Vue.use(VueRouter);

export const routes = [
  {
    path: "/",
    component: TvShowList,
    meta: {
      title: "Tv Show"
    }
  },
  {
    path: "/:showId",
    component: ShowDetail,
    name: "showDetail",
    meta: {
      title: "Show Detail"
    }
  },
  {
    path: "*",
    name: "PageNotFound",
    component: PageNotFound,
    meta: {
      title: "Page-not-found"
    }
  }
];

export const router = new VueRouter({
  mode: "history",
  base: process.env.BASE_URL,
  routes
});

router.afterEach(to => {
  document.title = to.meta.title || "Home";
});
