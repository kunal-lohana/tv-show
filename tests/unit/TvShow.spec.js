import {
  mount,
  shallowMount,
  config,
  createLocalVue,
  RouterLinkStub
} from "@vue/test-utils";
import TvShowComp from "@/components/TvShow/TvShow.vue";
import { homeShows } from "./testingData";
import Vuetify from "vuetify";
import Vue from "vue";
import VueRouter from "vue-router";
import { routes } from "@/router/index";
import { Constants } from "@/Constant/index";

config.showDeprecationWarnings = false;
Vue.config.slient = false;
describe("Tv Show Component", () => {
  const show = homeShows;
  const router = new VueRouter({
    routes
  });
  let wrapper;
  let localVue = createLocalVue();
  localVue.use(VueRouter);
  Vue.use(Vuetify);
  wrapper = mount(TvShowComp, {
    localVue,
    router,
    propsData: {
      showData: show
    },
    stubs: {
      RouterLink: RouterLinkStub
    }
  });

  it("should match snapshot", () => {
    const wrapper = shallowMount(TvShowComp, {
      localVue,
      router,
      propsData: {
        showData: show
      },
      stubs: {
        RouterLink: RouterLinkStub
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("check default value of data and props", () => {
    const wrapper = shallowMount(TvShowComp, {
      localVue,
      router,
      propsData: {
        showData: show
      },
      stubs: {
        RouterLink: RouterLinkStub
      }
    });
    expect(wrapper.vm.showData).toEqual(show);
    expect(wrapper.vm.showGenre).toBeTruthy();
    expect(wrapper.vm.constant).toEqual(Constants);
  });

  it("display TvShow name", () => {
    const name = wrapper.find("#data-test-tvshow-name").text();
    expect(name).toBe("Cops");
  });

  it("Display Rating", () => {
    const rating = wrapper.find("#data-test-tv-rating");
    expect(rating.text()).toEqual(homeShows.rating.average.toString());
  });
  it("Display Language", () => {
    const language = wrapper.find("#data-test-tv-language");
    expect(language.text()).toEqual(`, ${homeShows.language}`);
  });

  it("display genre when it is not belong to genre category", async () => {
    await wrapper.setProps({ showGenre: true });
    const genre = await wrapper.find("#data-test-tv-genre").text();

    expect(genre).toEqual(homeShows.genres.join(","));
  });

  it("check genre class when it is not belong to genre category", async () => {
    await wrapper.setProps({ showGenre: true });
    const genre = await wrapper.find("#data-test-genre-class").classes()[2];
    expect(genre).toEqual("show_genre");
  });

  it("genre class not exist, when it is belong to genre category", async () => {
    await wrapper.setProps({ showGenre: false });
    const genre = await wrapper.find("#data-test-genre-class").classes()[2];
    expect(genre).not.toEqual("show_genre");
  });

  it("Doesn not display genre when it is belong to genre category", async () => {
    await wrapper.setProps({ showGenre: false });
    const genre = wrapper.find("#data-test-tv-genre");
    expect(genre.exists()).toBeFalsy();
  });

  it("Display Premiere", () => {
    const premiered = wrapper.find("#data-test-tv-premiered");
    expect(premiered.text()).toEqual(`, ${homeShows.premiered.split("-")[0]}`);
  });

  it("check router-link props", () => {
    const wrapper = shallowMount(TvShowComp, {
      localVue,
      router,
      propsData: {
        showData: show
      },
      stubs: {
        RouterLink: RouterLinkStub
      }
    });
    expect(wrapper.find(RouterLinkStub).props().to).toEqual({
      name: "showDetail",
      params: { showId: 1 }
    });
  });

  it("check text of constant show detail button", () => {
    const wrapper = mount(TvShowComp, {
      localVue,
      router,
      propsData: {
        showData: show
      },
      stubs: {
        RouterLink: RouterLinkStub
      }
    });
    const constantPagefound = wrapper.find(
      '[data-test="constant-showDetailBtn"]'
    );
    expect(constantPagefound.text()).toEqual(
      wrapper.vm.constant.SHOW_DETAILS_BTN
    );
  });

  it("check image url ", () => {
    const wrapper = shallowMount(TvShowComp, {
      localVue,
      router,
      propsData: {
        showData: show
      },
      stubs: {
        RouterLink: RouterLinkStub
      }
    });
    const imageUrl = wrapper
      .find('[data-test="tv-show-image"]')
      .attributes("src");
    expect(imageUrl).toEqual(show.image.medium);
  });

  it("check Icon text", () => {
    const wrapper = shallowMount(TvShowComp, {
      localVue,
      router,
      propsData: {
        showData: show
      },
      stubs: {
        RouterLink: RouterLinkStub
      }
    });
    const rating = wrapper.find('[data-test="rating-icon"]');
    expect(rating.text()).toEqual(wrapper.vm.ratingIcon);
  });
});
