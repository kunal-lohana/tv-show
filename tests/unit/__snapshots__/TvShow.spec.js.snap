// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tv Show Component should match snapshot 1`] = `
<v-flex-stub tag="div" xs6="" sm4="" md3="" lg3="" class="pa-4">
  <v-card-stub loaderheight="4" tag="div" color="secondary" width="200" dark="true" app="" class="tv_show_card">
    <v-img-stub height="200" options="[object Object]" position="center center" src="asdfsd" transition="fade-transition" data-test="tv-show-image" class="white--text align-end">
      <v-card-title-stub id="data-test-tvshow-name">Cops</v-card-title-stub>
    </v-img-stub>
    <div>
      <v-card-text-stub id="data-test-genre-class" class=" 0 1">
        <!---->
        <div>
          <v-icon-stub color="amber" datatest="rating-icon" data-test="rating-icon">mdi-star</v-icon-stub>
          |
          <span id="data-test-tv-rating" class="pr-1">7</span> <span id="data-test-tv-language" class="pr-1">, English</span> <span id="data-test-tv-premiered" class="pr-1">, 1989</span>
        </div>
      </v-card-text-stub>
    </div>
    <v-card-actions-stub class="justify-center">
      <v-btn-stub rounded="true" tag="button" activeclass="" text="true" type="button" class="tvshow-detail-btn"><a text="" color="secondary" data-test="constant-showDetailBtn" class="text-decoration-none">
          Show Details
        </a></v-btn-stub>
    </v-card-actions-stub>
  </v-card-stub>
</v-flex-stub>
`;
